base/compiler/typelimits.jl

```diff
@@ -256,7 +256,7 @@ function type_more_complex(@nospecialize(t), @nospecialize(c), sources::SimpleVe
         return type_more_complex(t, c.a, sources, depth, tupledepth, allowed_tuplelen) &&
                type_more_complex(t, c.b, sources, depth, tupledepth, allowed_tuplelen)
     elseif isa(t, Int) && isa(c, Int)
-        return t !== 1 # alternatively, could use !(0 <= t < c)
+        return !(0 <= t < c)
```

Comment:
We can probably also keep `&& t !== 1`.

Or, if we felt like being even more permissive, we could do the comparison with `!(abs(t) <= abs(c) || t < n)`. (for some arbitrary value of `n >= 1`, such as `1`). Maybe just put that as the "# alternatively, could use" comment for the next person? 😜